src/base/http/connection.cpp

```diff
@@ -123,14 +135,30 @@ void Connection::read()
             {
                 const Environment env {m_socket->localAddress(), m_socket->localPort(), m_socket->peerAddress(), m_socket->peerPort()};
 
-                Response resp = m_requestHandler->processRequest(result.request, env);
+                if (result.request.method == HEADER_REQUEST_METHOD_HEAD)
+                {
+                    Request getRequest = result.request;
+                    getRequest.method = HEADER_REQUEST_METHOD_GET;
 
-                if (acceptsGzipEncoding(result.request.headers[u"accept-encoding"_s]))
-                    resp.headers[HEADER_CONTENT_ENCODING] = u"gzip"_s;
+                    Response resp = m_requestHandler->processRequest(getRequest, env);
 
-                resp.headers[HEADER_CONNECTION] = u"keep-alive"_s;
+                    resp.headers[HEADER_CONNECTION] = u"keep-alive"_s;
+                    resp.headers[HEADER_CONTENT_LENGTH] = QString::number(resp.content.length());
```

Comment:
Why does it return some "length" if there is no content?

Reply:
Please read the first paragraph:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD